src/cmd/root.go

```diff
@@ -0,0 +1,224 @@
+/*
+Copyright © 2019 Red Hat Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/containers/toolbox/pkg/podman"
+	"github.com/containers/toolbox/pkg/version"
+
+	"github.com/containers/toolbox/pkg/utils"
+	"github.com/spf13/cobra"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+)
+
+var (
+	cfgFile   string
+	rootFlags struct {
+		loglevel  string
+		logPodman bool
+		assumeyes bool
+		verbose   bool
+	}
+	rootCmd = &cobra.Command{
+		Use:   "toolbox",
+		Short: "Unprivileged development environment",
+		Long: `Toolbox is a tool that offers a familiar RPM based environment for
+developing and debugging software that runs fully unprivileged using Podman.`,
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
```

Comment:
Would be better to use a separate top-level function for this. If nothing else, the code wouldn't be so heavily indented.

Reply:
Right!